### PR TITLE
Feature #67: Update CRA template/React SDK to use new SDK version

### PR DESCRIPTION
**Pull request by mikheevm**
This PR:
- Updates CRA template/Safe Apps React SDK to the new version of the SDK
- Publish both packages under beta versions
- Rename SafeProvider `loading` prop to `loader`. Also, change the behavior of a loader. Now it's displayed only if loader prop is provided and the app is not connected.
- Export SDK opts from SDK package

Testing: 
```
npx create-react-app my-safe-app --template @gnosis.pm/cra-template-safe-app
```
Choose the latest beta version

### CODE REVIEW

File: packages/cra-template-safe-app/template/src/App.tsx [hunk 1]
```diff
@@ -15,29 +15,33 @@ const Container = styled.form`
...
       console.error(e);
     }
     setSubmitting(false);
-  }, [safe]);
+  }, [safe, sdk]);
+
   return (
     <Container>
-      <Title size="md">{safe.info.safeAddress}</Title>
+      <Title size="md">{safe?.safeAddress}</Title>
```
Comment (rmeissner) [inline packages/cra-template-safe-app/template/src/App.tsx:44]: shouldn't `safe` be always set? (looking at https://github.com/gnosis/safe-apps-sdk/pull/68/files#diff-301eef080af00a69a696e6909cdebd171073aae7d3eae63afaadb7682acbdfbcR5)

File: packages/safe-apps-react-sdk/src/index.tsx [hunk 2]
```diff
@@ -1,34 +1,53 @@
...
+  const [connected, setConnected] = useState(false);
+  const [safe, setSafe] = useState<SafeInfo>({ safeAddress: '', network: 'rinkeby' });
+  const contextValue = useMemo(() => ({ sdk, connected, safe }), [sdk, connected, safe]);
+
+  useEffect(() => {
+    const fetchSafeInfo = async () => {
+      try {
+        const safeInfo = await sdk.getSafeInfo();
+        setConnected(true);
+        setSafe(safeInfo);
```
Comment (rmeissner) [inline packages/safe-apps-react-sdk/src/index.tsx:28]: Should we first call `setSafe` and then `setConnected` ? that that the hook always returns the correct Safe info when it is connected? (Not sure if it actually makes a difference, as JS is not truly concurrent)

File: tsconfig.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
     "esModuleInterop": true /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */,
     "experimentalDecorators": true /* Enables experimental support for ES7 decorators. */,
     "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */,
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "jsx": "react-jsx"
```
Comment (rmeissner) [inline tsconfig.json:19]: Can we have a global tsconfig.json for the monorepo (at like a base file)?

**Review by rmeissner — APPROVED** [review]

File: tsconfig.json (hunk 3 above)
Comment (mikheevm) [reply tsconfig.json:19]: yes, this one is global and placed in the root

File: packages/safe-apps-react-sdk/src/index.tsx (hunk 2 above)
Comment (mikheevm) [reply packages/safe-apps-react-sdk/src/index.tsx:28]: yeah looks like it makes a difference: https://stackoverflow.com/questions/53048495/does-react-batch-state-update-functions-when-using-hooks and https://www.reddit.com/r/reactjs/comments/dobo9p/does_react_still_batch_state_updates_with/

File: tsconfig.json (hunk 3 above)
Comment (rmeissner) [reply tsconfig.json:19]: sorry ... didn't look where this file is located, just saw that we have the same definition in another folder 🤦 .... that is why you don'T do reviews in the middle of the night

**Pull request merged**